packages/cli/src/commands/onboard/auth/patches/01-github.Add-SignInPage.patch

```diff
@@ -0,0 +1,26 @@
+--- a/packages/app/src/App.tsx
++++ b/packages/app/src/App.tsx
+@@ -30 +30,5 @@ import { Root } from './components/Root';
+-import { AlertDisplay, OAuthRequestDialog } from '@backstage/core-components';
++import {
++  AlertDisplay,
++  OAuthRequestDialog,
++  SignInPage,
++} from '@backstage/core-components';
+@@ -35,0 +40 @@ import { catalogEntityCreatePermission } from '@backstage/plugin-catalog-common/
++import { githubAuthApiRef } from '@backstage/core-plugin-api';
+@@ -38,0 +44,14 @@ const app = createApp({
++  components: {
++    SignInPage: props => (
++      <SignInPage
++        {...props}
++        auto
++        provider={{
++          id: 'github-auth-provider',
++          title: 'GitHub',
++          message: 'Sign in using GitHub',
++          apiRef: githubAuthApiRef,
++        }}
++      />
++    ),
++  },
```

Comment:
Just wanna call out that this as well as the sign-in resolver are things that we want to figure out how to more or less eliminate from default code configuration, i.e. make them completely config driven instead.

No action required on this PR but just highlighting that we shouldn't invest large amounts of time into making this code configuration experience better. At that point it's better to start up the work to make this config driven instead.

Reply:
Actually, do you think it's worth considering moving that into scope of this effort? I do feel that a close to zero-config sign-in system would significantly simplify adoption.

Reply:
> just highlighting that we shouldn't invest large amounts of time into making this code configuration experience better.

Hey, I just wanted to chip in a bit here with extra context.
My understanding is that this "onboard" command is trying to 80% reduce onboarding friction with 20% of the effort.

I absolutely agree that the _correct_ way of solving this problem is identifying each manual (or code-change) step of onboarding, then evaluating if it _should_ be manual (or config instead), or how it can otherwise be improved. However, it's a relatively high cost to change each of these:
* It's a breaking change
* We'll want to be relatively confident for each one of these changes, to avoid having to tweak them again
* There may be migration needs, etc

Fortunately, I think the "onboard" command _will_ help with improving the underlying changes: due to its experimental nature and "one-off, 'patch or bail'" approach, we have a lot of flexibility to iterate and innovate with it.
Over time, I'm expecting it to organically "smooth down" and expose a _very_ ergonomic UX to users, at which point we can learn from it to much more confidently and easily improve the underlying config/code-change requirements.

-----

In summary, I'd advocate towards deferring the work to improve the underlying config/code-change requirements for two reasons:
1. We get 80% of the UX experience without needing nearly as much planning work up-front
2. We still get to improve the underlying experience in the future, but we'll be equipped with the learnings from the "onboard" command.

Reply:
Yep! Just being mindful of spending too much time polishing a 💩, 😁